main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// TODO: add language flag diversion, unit unit test
+
+func main() {
+	// peeper subcommand "get"
+	getCmd := flag.NewFlagSet("get", flag.ExitOnError)
+	// Flags for "get" subcommand
+	// ("flag name", default, "Flag explanation")
+	getAll := getCmd.Bool("all", false, "Get full shortcut list")
+	getByKeyword := getCmd.String("keyword", "", "Find a shortcut with keyword ")
+
+	// peeper subcommand "add"
+	addCmd := flag.NewFlagSet("add", flag.ExitOnError)
+	// Flags(inputs) for "add" subcommand
+	addName := addCmd.String("name", "", "Name of the shortcut")
+	addShortcut := addCmd.String("shortcut", "", "Shortcut")
+
+	// peeper subcommand "delete"
+	deleteCmd := flag.NewFlagSet("delete", flag.ExitOnError)
+	// Flags for "delete" subcommand
+	deleteAll := deleteCmd.Bool("all", false, "Delete whole shortcut list")
+	deleteByName := deleteCmd.String("name", "", "Delete a shortcut by name")
+
+	if len(os.Args) < 2 {
+		fmt.Println("Exepected either 'get', 'add', 'delete' subcommands")
+	}
+
+	switch os.Args[1] {
+	case "get":
+		HandleGet(getCmd, getAll, getByKeyword)
+	case "add":
+		HandleAdd(addCmd, addName, addShortcut)
+	case "delete":
+		HandleDelete(deleteCmd, deleteAll, deleteByName)
+		// TODO: Do we need to add default and return error for the case of the command doesn't exist?
+	}
+}
+
+func HandleGet(getCmd *flag.FlagSet, all *bool, keyword *string) {
+	getCmd.Parse(os.Args[2:])
+
+	if *all == false && *keyword == "" {
+		fmt.Println("Specify the target shortcut with all or keyword flag")
+		getCmd.PrintDefaults()
+		os.Exit(1)
+	}
+
+	if *all {
+		// Return full shortcut list
+		shortcuts := getShortcuts()
+		fmt.Println("Name \t Shortcut key \n")
+		for _, shortcut := range shortcuts {
+			fmt.Printf("%v \t %v \n", shortcut.Name, shortcut.ShortcutKey)
+		}
+		return
+	}
+
+	if *keyword != "" {
+		shortcuts := getShortcuts()
+		keyword := *keyword
+		for _, shortcut := range shortcuts {
+			if strings.Contains(shortcut.ShortcutKey, keyword) {
+				fmt.Println("Name \t Shortcut key \n")
+				fmt.Printf("%v \t %v \n", shortcut.Name, shortcut.ShortcutKey)
+			}
+		}
+	}
+}
+
+func ValidateNewShortcutKey(addCmd *flag.FlagSet, name *string, shortcut *string) {
+	addCmd.Parse(os.Args[2:])
+	if *name == "" || *shortcut == "" {
+		fmt.Print("Name and the shortcut key are required to add a shortcut key")
+		addCmd.PrintDefaults()
+		os.Exit(1)
```

Comment:
Can you refactor your code to return errors instead, and only have a single os.Exit(1) in your `main` function?

[Here's a blog post with a pattern that I like that might help you](https://pace.dev/blog/2020/02/12/why-you-shouldnt-use-func-main-in-golang-by-mat-ryer.html), or more concretely [my own template](https://github.com/kinbiko/go-template/blob/9f171de74003e153e12ba72df93c5afc50cd20f1/cmd/app/main.go#L8-L17) with this pattern.

Reply:
Thanks for the reference!
Will update this part later.

Reply:
Updated in the change [cb47f77](https://github.com/snkzt/shortcut-peeper/pull/2/commits/cb47f7760c67ccf8b8f138393a4d9a65ed4b7f71)!